src/url_util.ts

```diff
@@ -370,3 +370,26 @@ export function interpolateSearchItem(urlPattern: URL, query: string): URL {
         return new URL(urlPattern.href + query)
     }
 }
+
+/**
+ * Check if url URL filter criteria matches the current page URL
+ *
+ * @param part      a URL part string or full URL be matched with the @param url
+ * @param url       a URL object of the current page
+ *
+ * @return          `true` if url `part` matches the full URL href, and `false` otherwise
+ *
+ */
+export function partMatchesURL(part: string, url: URL): boolean {
```

Comment:
I don't understand the need for this over `String.prototype.search`.

Did Vimperator have something similar?